src/content/lodash/util/cond/vanilla.js

```diff
@@ -0,0 +1,22 @@
+const cond = (pairs) => (item) => {
+  for (const [predicate, fn] of pairs) {
+    if (predicate(item)) {
+      return fn(item);
+    }
+  }
+};
```

Comment:
```suggestion
const cond = (pairs) => (...args) => {
  for (const [predicate, fn] of pairs) {
    if (predicate(...args)) {
      return fn(...args);
    }
  }
};
```

It seems that `_.cond` doesn't limit the amount of arguments in predicates/functions

Reply:
If you test this, it doesn't actually work that way. If we want to add that functionality, we can, but we should document it as an enhancement and not a replacement.

see [this codepen](https://codepen.io/SgtPooki/pen/oNwaLaY?editors=0011). Code copied below to prevent data-loss:

```javascript
// for https://github.com/cedmax/youmightnotneed/pull/156#discussion_r688319534
import * as _ from "https://cdn.skypack.dev/lodash@4.17.21";

// from https://lodash.com/docs/4.17.15#cond
var func = _.cond([
  [_.matches({ a: 1 }), _.constant("matches A1")],
  [_.conforms({ b: _.isNumber }), _.constant("matches B1")],
  [_.stubTrue, _.constant("no match1")]
]);

var log = (...args) => console.log(...args);
console.clear();

const runTest = (fn, msg) => {
  log(`START: ${msg}`);

  log("First lodash example");
  log(fn({ a: 1, b: 2 }));
  // => 'matches A'

  log("Second lodash example");
  log(fn({ a: 0, b: 1 }));
  // => 'matches B'

  log("Third lodash example");
  log(fn({ a: "1", b: "2" }));
  // => 'no match'

  // only the first item is processed...
  log("Pass multiple arguments to check for rest argument support");
  log(fn({ a: "1", b: "2" }, { a: 0, b: 1 }, { a: 1, b: 2 }));
  log(`END: ${msg}`);
};

var funcWithoutDefault = _.cond([
  [_.matches({ a: 1 }), _.constant("matches A1")],
  [_.conforms({ b: _.isNumber }), _.constant("matches B1")]
]);

runTest(func, `Using func with "no match" catch-all predicate \`_.stubTrue\``);
runTest(
  funcWithoutDefault,
  `Using funcWithoutDefault function to test if rest parameters are processed differently when there is no default`
);

/**
Console output: 
"START: Using func with 'no match' catch-all predicate `_.stubTrue`"
"First lodash example"
"matches A1"
"Second lodash example"
"matches B1"
"Third lodash example"
"no match1"
"Pass multiple arguments to check for rest argument support"
"no match1"
"END: Using func with 'no match' catch-all predicate `_.stubTrue`"
"START: Using funcWithoutDefault function to test if rest parameters are processed differently when there is no default"
"First lodash example"
"matches A1"
"Second lodash example"
"matches B1"
"Third lodash example"
undefined
"Pass multiple arguments to check for rest argument support"
undefined
"END: Using funcWithoutDefault function to test if rest parameters are processed differently when there is no default"

*/
```

Reply:
If each of those items were processed, then 1. I feel like lodash docs would mention it, 2. I should get an array of results

Reply:
I think we have misunderstood each other, here is my example where is the replacement implementation differs from `_.cond` implementation:

```js
const pairs = [
  [(x, y) => x > y, (x, y) => x - y],
  [() => true, (x, y) => y - x],
];

const f0 = _.cond(pairs);

const cond = (pairs) => (item) => {
  for (const [predicate, fn] of pairs) {
    if (predicate(item)) {
      return fn(item);
    }
  }
};

const f1 = cond(pairs);

// cond from lodash
f0(0, 1);
// => 1
f0(0, 5);
// => 5

// cond from this PR
f1(0, 5);
// => NaN
f1(0, 1);
// => NaN
```

My suggestion fixes this and only this behavior

Reply:
Updated. Resolving this thread.